packages/terra-application/src/primary-navigation-layout/workspace-layout/PrimaryNavigationWorkspace.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Workspace, { WorkspaceItem } from '../../workspace';
+
+const propTypes = {
+  /**
+   * Callback function triggering when the workspace size changes.
+   */
+  onSizeChange: PropTypes.func,
+  /**
+   * Callback function triggering when the presentation state changes.
+   */
+  onPresentationStateChange: PropTypes.func,
+  /**
+   * Callback function triggering when the active.
+   */
+  onActiveItemChange: PropTypes.func,
+  /**
+   * The string key linked to the action workspace.
+   */
+  initialActiveItemKey: PropTypes.string.isRequired,
+  /**
+   * Child content to be placed within the main content region.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Whether or not the workspace should initially display as open.
+   */
+  initialIsOpen: PropTypes.bool,
+  /**
+   * The size string value matching the active size option.
+   */
+  initialSize: PropTypes.object,
+  /**
+   * @private
+   * Id string to apply to the workspace
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * @private
+   * Whether or not the workspace is open
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * @private
+   * Function callback i.e. `onRequest(event)`
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * @private
+   * Whether or not the workspace is present as an overlay
+   */
+  isPresentedAsOverlay: PropTypes.bool,
+  /**
+   * @private
+   * Numeric scale value ranging from `0.0 - 1.0` as the minimum to maximum size for the workspace
+   */
+  sizeScalar: PropTypes.number,
+  /**
+   * @private
+   * The string representation of the workspace size
+   */
+  activeSize: PropTypes.string,
+  /**
+   * @private
+   * Array of objects containing key/text pairs for the available size options
+   */
+  sizeOptions: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+  })),
+  /**
+   * @private
+   * Function callback i.e. `onRequestSizeChange(size)`
+   */
+  onRequestSizeChange: PropTypes.func,
+};
+
+const WorkspaceWrapper = ({
+  // consumer props
+  onSizeChange,
+  onPresentationStateChange,
+  onActiveItemChange,
+  initialActiveItemKey,
+  children,
+  initialIsOpen, // eslint-disable-line no-unused-vars, react/no-unused-prop-types
```

Comment:
are these unused props placeholders for an upcoming api update?

Reply:
These props are read by the parent to manage initial state, but the isOpen proper etc, is meant to pass down.  I'll add a comment.

Reply:
If they're unused below, you could just remove them from the destructuring here. We're not doing custom props, so we don't have to worry about stripping them out of anything. You would have to move the no-unused-prop-type override up into the props table tho.